nilearn/plotting/displays.py

```diff
@@ -237,7 +245,24 @@ def transform_to_2d(self, data, affine):
                 The affine of the volume
 
         """
-        max_axis = 'xyz'.index(self.direction)
+        if self.direction in 'xlr':
+            max_axis = 0
+        else:
+            max_axis = '.yz'.index(self.direction)
+
+        # set unselected brain hemisphere activations to 0
+        if self.direction == 'l':
+            data_selection = data.copy()
+            x_center, _, _, _ = np.dot(np.linalg.inv(affine),
+                                       np.array([0, 0, 0, 1]))
+            data_selection[int(x_center):, :, :] = 0
```

Comment:
This will not work if the affine is not diagonal. Is that always the case? If so, you could just do `data_selection=data[:int(x_center), :, :]` to avoid the copy above.